python/hail/docs/scans.rst

```diff
@@ -0,0 +1,78 @@
+.. _sec-scan:
+
+Scans
+===========
+
+The ``scan`` module is exposed as ``hl.scan``, e.g. ``hl.scan.sum``.
+
+The functions in this module perform rolling aggregations along the desired axis;
+For example, the ``count`` aggregator can be used as ``hl.scan.count`` to add an
+index along the rows of a `Table` or the rows or columns of a `MatrixTable`.
+
+For example, to add a rolling sum of a Table field, we could do:
+
+    >>> ht_scan = ht.select(ht.ID, ht.Z, rolling_sum=hl.scan.sum(ht.Z))
+    >>> ht_scan.show()
+    +-------+-------+-------------+
+    |    ID |     Z | rolling_sum |
+    +-------+-------+-------------+
+    | int32 | int32 |       int64 |
+    +-------+-------+-------------+
+    |     1 |     4 |           0 |
+    |     2 |     3 |           4 |
+    |     3 |     3 |           7 |
+    |     4 |     2 |          10 |
+    +-------+-------+-------------+
+
+The corresponding calls on MatrixTable fields would be:
```

Comment:
"To compute the cumulative number of non-reference genotype calls along the genome:"

And name the field something like `cum_n_non_ref`